components/director/internal/open_resource_discovery/accessstrategy/cmp_mtls_access_strategy.go

```diff
@@ -0,0 +1,92 @@
+package accessstrategy
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+	"sync"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/cert"
+	"github.com/pkg/errors"
+	"github.com/vrischmann/envconfig"
+)
+
+const cmpMTLSConfigPrefix = "ACCESS_STRATEGY_SAP_CMP_MTLS_V1"
+
+type cmpMTLSAccessStrategyExecutor struct {
+	lock sync.RWMutex
+
+	client *http.Client
+
+	config *cert.CertSvcConfig
+}
+
+// NewCMPmTLSAccessStrategyExecutor creates a new Executor for the CMP mTLS Access Strategy
+func NewCMPmTLSAccessStrategyExecutor() *cmpMTLSAccessStrategyExecutor {
+	return &cmpMTLSAccessStrategyExecutor{
+		lock: sync.RWMutex{},
+	}
+}
+
+// Execute performs the access strategy's specific execution logic
+func (as *cmpMTLSAccessStrategyExecutor) Execute(ctx context.Context, baseClient *http.Client, documentURL string) (*http.Response, error) {
+	if !as.isInitialized() {
+		if err := as.initialize(ctx, baseClient); err != nil {
+			return nil, errors.Wrap(err, "while initializing access strategy sap:cmp-mtls:v1")
+		}
+	}
+	return as.client.Get(documentURL)
+}
+
+// SetConfig sets the Access Strategy config. This is used when reading from environment is not suitable in that context.
+func (as *cmpMTLSAccessStrategyExecutor) SetConfig(config cert.CertSvcConfig) {
```

Comment:
thats a weird one - whats the purpose ? usually if you have to reconfigure something during runtime, its better to instantiate a new object every time